crates/turborepo-lib/src/engine/builder.rs

```diff
@@ -80,7 +80,7 @@ impl<'a> EngineBuilder<'a> {
     pub fn with_root_tasks<I: IntoIterator<Item = TaskName<'static>>>(mut self, tasks: I) -> Self {
         self.root_enabled_tasks = tasks
             .into_iter()
-            .filter(|name| name.package() == Some(ROOT_PKG_NAME))
+            .filter(|name| name.package() == Some(ROOT_PKG_NAME) || name.package().is_none())
```

Comment:
I don't think this is right, we're now considering all tasks that aren't workspace tasks as root tasks:
e.g. a pipeline with entries for just `build`, `test` and `//#test` we're now marking both `build` and `test` as root tasks when they aren't. 

This is bad because we now are listing `//#build` as a task even if the user hasn't specified it as a task which goes against our docs: 
> For example, if your pipeline declares a "build" task, and you want to include the build script defined in the monorepo's root package.json file with turbo run build, you must opt the root into it by declaring "//#build": {...} in your configuration

[link](https://turbo.build/repo/docs/core-concepts/monorepos/running-tasks#running-tasks-from-the-root)

Reply:
Yeah you're right. I was confused as to what having `package` be `None` meant